src/mbed_tools/project/_internal/project_data.py

```diff
@@ -114,13 +115,21 @@ def from_existing(cls, root_path: Path) -> "MbedProgramFiles":
         cmakelists_file = root_path / CMAKELISTS_FILE_NAME
         if not cmakelists_file.exists():
             logger.warning("No CMakeLists.txt found in the program root.")
+        if build_subdir is None:
```

Comment:
Are we allowing build_subdir to be None? What does it mean when build_subdir is None? I thought we wanted to always write to a target-specific subdirectory. This looks like it it making a new "None" folder sometimes.

Reply:
I added None as an option so that tests would be able to check that from_existing and from_new will produce the same MbedProgramFiles, since from_new doesn't take target and toolchain arguments (and so doesn't point to a configuration specific build_directory). It shouldn't ever create the "None" subdirectory as the build_subdir argument is always used outside of tests, but I can remove the / "None" to reduce confusion.

Reply:
Maybe there is another way to test this, perhaps with mocking. Then we wouldn't need to add "test only" code to our production Python files (just the test ones).